models/User.model.js

```diff
@@ -1 +1,23 @@
-// User model here
+const { Schema, model } = require('mongoose');
+
+const userSchema = new Schema (
+  {
+    username: {
+      type: String,
+      required: true,
+      match: [/^\S+@\S+\.\S+$/, 'Please use a valid email address.'],
```

Comment:
Nice thought with the validation implementation here. However, it looks like you're validating the username to have the same format as an email. This will cause issues when a user is signing up. 

The application asks for a user to choose a username and not register with their email. When revisiting your solution, you can remove this matcher.